packages/amplify-provider-awscloudformation/src/amplify-service-manager.js

```diff
@@ -249,107 +249,117 @@ async function deleteEnv(context, envName, awsConfigInfo) {
+  const amplifyMeta = stateManager.getMeta();
+  const providerMeta = amplifyMeta.providers[ProviderName];
 
-  const { envName } = localEnvInfo;
-  const stackName = teamProviderInfo[envName][ProviderName].StackName;
-  const region = teamProviderInfo[envName][ProviderName].Region;
+  const appId = providerMeta[AmplifyAppIdLabel];
 
-  if (!teamProviderInfo[envName][ProviderName][AmplifyAppIdLabel]) {
-    let amplifyAppId;
-
-    const amplifyClient = await getConfiguredAmplifyClient(context);
-    if (!amplifyClient) {
-      // This happens when the Amplify service is not available in the region
-      return;
-    }
+  if (appId) {
+    return;
+  }
 
-    const hasPermission = await checkAmplifyServiceIAMPermission(context, amplifyClient);
-    if (!hasPermission) {
-      return;
-    }
+  const stackName = providerMeta.StackName;
+  const region = providerMeta.Region;
 
-    const searchAmplifyServiceResult = await searchAmplifyService(amplifyClient, stackName);
+  let amplifyAppId;
 
-    if (searchAmplifyServiceResult.backendEnvExists) {
-      amplifyAppId = searchAmplifyServiceResult.amplifyAppId; // eslint-disable-line
-    } else {
-      const envList = Object.keys(teamProviderInfo);
+  const amplifyClient = await getConfiguredAmplifyClient(context);
+  if (!amplifyClient) {
+    // This happens when the Amplify service is not available in the region
+    return;
+  }
 
-      let appIdsInTheSameLocalProjectAndRegion = [];
-      for (const env of envList) {
-        if (
-          env !== envName &&
-          teamProviderInfo[env][ProviderName].Region === region &&
-          teamProviderInfo[env][ProviderName][AmplifyAppIdLabel]
-        ) {
-          appIdsInTheSameLocalProjectAndRegion.push(teamProviderInfo[env][ProviderName][AmplifyAppIdLabel]);
-        }
-      }
+  const hasPermission = await checkAmplifyServiceIAMPermission(context, amplifyClient);
+  if (!hasPermission) {
+    return;
+  }
 
-      const verifiedAppIds = searchAmplifyServiceResult.apps.map(app => app.appId);
-      appIdsInTheSameLocalProjectAndRegion = appIdsInTheSameLocalProjectAndRegion.filter(appId => verifiedAppIds.includes(appId));
+  const searchAmplifyServiceResult = await searchAmplifyService(amplifyClient, stackName);
 
-      if (appIdsInTheSameLocalProjectAndRegion.length === 1) {
-        amplifyAppId = appIdsInTheSameLocalProjectAndRegion[0]; // eslint-disable-line
-      } else if (appIdsInTheSameLocalProjectAndRegion.length > 1) {
-        context.print.info(`Your project is associated with multiple Amplify Service Apps in the region ${region}`);
-        amplifyAppId = await SelectFromExistingAppId(context, appIdsInTheSameLocalProjectAndRegion);
+  if (searchAmplifyServiceResult.backendEnvExists) {
+    amplifyAppId = searchAmplifyServiceResult.amplifyAppId; // eslint-disable-line
+  } else {
+    // TODO reimplement this check using service calls instead of team-provider-info.json
```

Comment:
I see TPI read from and written to here and elsewhere in this file. Is that a blocker?

Reply:
not a blocker. The next phase of the refactor will be abstracting the provider block into service calls and reading/writing to `amplify-meta.json`instead